frontend/middleware_validatesubscription.go

```diff
@@ -49,6 +54,13 @@ func (s *SubscriptionStateMuxValidator) MiddlewareValidateSubscriptionState(w ht
 		return
 	}
 
+	// Emit the subscription state metric
+	s.metrics.EmitGauge("subscription_lifecycle", 1, map[string]string{
```

Comment:
the name `subscription_lifecycle` implies this is emitted when the Subscription Lifecycle endpoint is called. However, this middleware is not invoked at that time. Rather, this middleware is invoked for every call a user would make on the resources provided.

perhaps a name like `resource_request_subscription_state`